pkg/upgrade/upgrade.go

```diff
@@ -179,53 +172,39 @@ func (ci *clusterInfo) updateInfo(info map[string]NodeVersion, dtk registry.Driv
 	return info, nil
 }
 
-func (ci *clusterInfo) driverToolkitVersion(ctx context.Context, entries []string, info map[string]NodeVersion) (map[string]NodeVersion, error) {
-	if len(entries) == 0 {
+func (ci *clusterInfo) driverToolkitVersion(ctx context.Context, dtkImages []string, info map[string]NodeVersion) (map[string]NodeVersion, error) {
+	if len(dtkImages) == 0 {
 		return info, nil
 	}
 
-	var dtk registry.DriverToolkitEntry
-	var imageURL string
+	var dtk *registry.DriverToolkitEntry
+	imageURL := dtkImages[0]
 
-	entry := entries[0]
-
-	if cached := ci.cache[entry]; cached != nil {
-		dtk = cached.DTK
-		imageURL = cached.ImageURL
-		ci.log.Info("History from cache", "entry", entry, "dtk", dtk, "imageURL", imageURL)
+	if cached := ci.cache[imageURL]; cached != nil {
+		dtk = cached
```

Comment:
So now I think we can do this:
```suggestion
	if dtk = ci.cache[imageURL]; dtk != nil {
```

Reply:
haha, absolutely :D